tests/analysis_agent.py

```diff
@@ -0,0 +1,176 @@
+        """
+        query_dict = {
+            'id': uuid,
+            'additionalQueryResultFields': ['labels']
+        }
+        
+        query_dict['label']['caas-collection-name'] = self.cromwell_collection
+
+        response = cwm_api.query(query_dict=query_dict, auth=self.auth)
+        response.raise_for_status()
+        result = response.json()
+        all_workflows = result['results']
+        total_count = result['totalResultsCount']
+        assert len(all_workflows) == total_count == 1
+        return Workflow(all_workflows[0])
+
+    def query_by_bundle(self, bundle_uuid, bundle_version=None, with_labels=True):
+        """Query the analysis workflows by their workflow-UUID.
+
+        Note, due to the open issue: https://github.com/broadinstitute/cromwell/issues/3115, if the result of workflows
+        are more than ~1000, this function will very likely raise an error.
+
+        Args:
+            bundle_uuid (str): HCA DCP bundle UUID.
+            bundle_version (str): Optional, HCA DCP bundle version. By default, it's None.
+            with_labels (bool): Optional, whether to return workflow labels in the results. By default, it's False.
+
+        Returns:
+            List[Workflow]: A list of Workflow objects. E.g. [Workflow_1, ..., Workflow_100]
+
+        Raises:
+            requests.exceptions.HTTPError: When the request to Secondary-analysis service (Cromwell) failed.
+        """
+        query_dict = {
+            'label': {
+                'bundle-uuid': bundle_uuid
+            }
+        }
+
+        if with_labels:
+            query_dict['additionalQueryResultFields'] = ['labels']
+
+        if bundle_version:
+            query_dict['label']['bundle-version'] = bundle_version
+        
+        query_dict['label']['caas-collection-name'] = self.cromwell_collection
+
+        response = cwm_api.query(query_dict=query_dict, auth=self.auth)
+        response.raise_for_status()
+        result = response.json()
+        all_workflows = [Workflow(wf) for wf in result['results']]
+        return all_workflows
+
+    def query_by_project_uuid(self, project_uuid, with_labels=True):
+        """Query the analysis workflows by the HCA DCP Ingest submission project-UUID, which is essentially one of the
+            workflow labels.
+
+        Note, due to the open issue: https://github.com/broadinstitute/cromwell/issues/3115, if the result of workflows
+        are more than ~1000, this function will very likely raise an error. The `with_labels` is a flag controlling the
+        behavior of whether to query the workflows asking for the labels in the response, by default it's set to True,
+        so please set it to False if you don't want to risk getting error responses.
+
+        Args:
+            project_uuid (str): HCA DCP Ingest submission project-UUID.
+            with_labels (bool): Optional, whether to return workflow labels in the results. By default, it's False.
```

Comment:
The default value is set to `True` in the function signature

Reply:
Argh 🤦‍♂️ Changed